calc.js

```diff
@@ -0,0 +1,32 @@
+function getFinalResult() {
+	const pairs = [90, 10, 1, 2, 3, 4, 5, 10, 1, 90, 90, 90, 10, 22];
+	let arrOfCheckedValues = []; // wrzucamy tu wartości bez dubli
+	let arrOfObjects = []; // wrzucamy tu object {id, value} bez dubli
+	let finalResult; // ostateczny wynik, liczba
+
+	for (let i = 0; i < pairs.length; i++) {
+
+		if (arrOfCheckedValues.indexOf(pairs[i]) === -1) {
+			// filtrujemy wskazaną wartość na tablicy
+			const value = pairs.filter(item => item === pairs[i]); 
+
+			// wrzucamy na nową tablicę objekt bez dubli
+			arrOfObjects.push({ id:pairs[i], value: value.length }); 
+
+			// wrzucamy wartość którą już sprawdziliśmy przefiltrowaliśmy
+			arrOfCheckedValues.push(pairs[i]); 
+		}
+	}
```

Comment:
To podejście ma sens, ale jest zapis który przydaje się w usuwanei duplikatów tj:
pairs.filter((element,index,array2) => array2.indexOf(element) === index)